tests/string.in

```diff
@@ -116,6 +116,8 @@ string repeat -n10 -m4 'foo'
 
 string repeat -n10 --max 5 'foo'
 
+string repeat -n3 -m20 'foo'
```

Comment:
This, as well as your other, unit tests is excellent. The only thing I would suggest is adding a few more "off by one" tests since that is a very common implementation bug. For example, `string repeat -n3 -m9 foo`. But perhaps your other tests already cover that case. In which case a few comments about what is being tested, and the expected outcome, would be helpful.